pkg/generators/corsproxy/generator.go

```diff
@@ -19,53 +21,84 @@ const (
 
 // Generator configures the generators for CORSProxy
 type Generator struct {
-	generators.BaseOptions
+	generators.BaseOptionsV2
 	Spec    saasv1alpha1.CORSProxySpec
 	Options config.Options
+	Traffic bool
 }
 
 // NewGenerator returns a new Options struct
-func NewGenerator(instance, namespace string, spec saasv1alpha1.CORSProxySpec) Generator {
+func NewGenerator(instance, namespace string, spec saasv1alpha1.CORSProxySpec) (Generator, error) {
```

Comment:
This function can never return an error, so I would remove the error from the function outputs.

Reply:
I left it like this to be consistent with https://github.com/3scale-ops/saas-operator/blob/main/pkg/generators/echoapi/generator.go#L26, which are prepared to maybe add the canary in the future (with even having already the send-traffic selector label...).

Better to remove it from both generators not having still the canary? (echoapi and corsproxy)? @roivaz

Reply:
Fixed https://github.com/3scale-ops/saas-operator/pull/157/commits/34ecee62ac2a899cae48b0caf54009b9a6fd6e08